### PR TITLE
Only show access token error if Mapbox data is used

### PR DESCRIPTION
**Pull request by Pessimistress**
Only warn about access token in console & canvas upon an unauthorized attempt to access Mapbox service.
This will eliminate the confusing message for non-mapbox-data users such as https://github.com/uber/react-map-gl/issues/385

### CODE REVIEW

File: src/components/static-map.js [hunk 1]
```diff
@@ -174,16 +178,25 @@ export default class StaticMap extends PureComponent {
     this._mapboxMap = ref;
   }
 
+  // Handle map error
+  _mapboxMapError(evt) {
+    if (evt.error && evt.error.status === 401 && !this.state.accessTokenInvalid) {
```
Comment (abmai) [inline src/components/static-map.js]: pull `401` out into a constant for better readability if it only signifies one thing? i.e.
```js
const UNAUTHORIZED_ERROR_CODE = 401;
```

File: src/mapbox/mapbox.js [hunk 2]
```diff
@@ -39,6 +39,7 @@ const propTypes = {
   attributionControl: PropTypes.bool, /** Show attribution control or not. */
   preserveDrawingBuffer: PropTypes.bool, /** Useful when you want to export the canvas as a PNG. */
   onLoad: PropTypes.func, /** The onLoad callback for the map */
+  onError: PropTypes.func, /** The onError callback for the map */
```
Comment (abmai) [inline src/mapbox/mapbox.js]: Outside the scope of this, but do we want to allow external overrides / ability to hook into these?

Comment (Pessimistress) [reply src/mapbox/mapbox.js]: Mapbox's full list of events (removed mouse/gesture events):
- resize
- remove
- webglcontextlost
- webglcontextrestored
- load
- render
- error
- data
- styledata
- sourcedata
- dataloading
- styledataloading
- sourcedataloading

Now that we expose `getMap` as a public method, the user is able to register their own listeners if they need to. Aside from `load` and `error`, I've also used `render` and `dataloading`. Not sure if all of them are useful in the React context.

Comment (abmai) [reply src/mapbox/mapbox.js]: That makes sense. Would registering them on the Mapbox instance override our event listener for those events, though?

Not a blocking thing for this PR btw, feel free to land whenever you want.

**Pull request merged**